randovania/game_connection/connector/dread_remote_connector.py

```diff
@@ -196,3 +196,7 @@ async def receive_remote_pickups(self) -> None:
 
         await self.executor.run_lua_code(execute_string)
         return
+
+    async def display_arbitrary_message(self, message: str):
+        execute_string = f"Game.AddSF(0, 'Scenario.QueueAsyncPopup', 'si', '{message}', 10.0)"
```

Comment:
shouldn't it properly wrap message, so it doesn't break if it contains a `'`?

Reply:
Oops, yeah, Just `escaped_message = message.replace("'", "\\'")` to escape the single quotes or is there a fancy python way?

Reply:
Can't we grab the lua_util from OPR?

Reply:
OPR? :p
Maybe I have a brain lag but ODR is just doing this:
```
def lua_convert(data, wrap_strings: bool = False) -> str:
    if isinstance(data, list):
        return "{\n" + "\n".join(
            f"{lua_convert(item, wrap_strings)},"
            for item in data
        ) + "\n}"
    if isinstance(data, dict):
        return "{\n" + "\n".join(
            f"{key} = {lua_convert(value, wrap_strings)},"
            for key, value in data.items()
        ) + "\n}"
    if isinstance(data, bool):
        return "true" if data else "false"
    if data is None:
        return "nil"
    if wrap_strings and isinstance(data, str):
        return wrap_string(data)
    return str(data)

def wrap_string(data: str) -> str:
    return f'"{data}"'
```
If I wrap the message with that, a single quote would still break it.

Reply:
Too many letters.

Hm... yeah, it would hm. I guess it'd also break multiworld stuff too, so I'm glad I restricted world names lol

Reply:
Just escaping `'` should work for now